test/microbench/org/apache/cassandra/test/microbench/DeletionTimeDeSerBench.java

```diff
@@ -0,0 +1,363 @@
+                buffer.clear();
+                if (dts.get(i).equals(DeletionTime.LIVE))
+                {
+                    byte b = buffer.get();
+                    b = (byte) (b & 1);
+                    bh.consume(b);
+                }
+                else
+                {
+                    // The flag
+                    byte flag = buffer.get();
+
+                    // 7 bytes mfda
+                    int bytes4 = buffer.getInt();
+                    int bytes2 = buffer.getShort();
+                    int bytes1 = buffer.get();
+
+                    long mfda = bytes4 & 0xFFFFFFFFL;
+                    mfda = (mfda << 16) + (bytes2 & 0xffff);
+                    mfda = (mfda << 8) + (bytes1 & 0xff);
+
+                    // The ldt
+                    int ldt = buffer.getInt();
+
+                    bh.consume(flag);
+                    bh.consume(bytes4);
+                    bh.consume(bytes2);
+                    bh.consume(bytes1);
+                    bh.consume(mfda);
+                    bh.consume(ldt);
+                }
+            }
+        }
+        else
+        {
+            for (int i = 0, m = dts.size(); i < m; i++)
+            {
+                buffer.clear();
+                long mfda = buffer.getLong();
+                int ldt = buffer.getInt();
+
+                bh.consume(mfda);
+                bh.consume(ldt);
+            }
+        }
+
+        bh.consume(dts);
+        bh.consume(buffer);
+    }
+
+    private DataInputPlus getDeSerIn(String value, ByteBuffer maybeBackingBB)
+    {
+        switch (value)
+        {
+            case RAMParam:
+                maybeBackingBB.rewind();
+                return new DataInputBuffer(maybeBackingBB, false);
+
+            case diskParam:
+                try
+                {
+                    return new FileInputStreamPlus(new org.apache.cassandra.io.util.File(diskFile));
+                }
+                catch(NoSuchFileException e)
+                {
+                    throw new RuntimeException(e);
+                }
+
+            default:
+                throw new IllegalStateException("Unknown backing media for deser.");
+        }
+    }
+
+    private DataOutputStreamPlus getSerOut(String value, ByteBuffer maybeBackingBB)
+    {
+        switch (value)
+        {
+            case RAMParam:
+                maybeBackingBB.rewind();
+                return new DataOutputBuffer(maybeBackingBB);
+
+            case diskParam:
+                try
+                {
+                    return new FileOutputStreamPlus(new org.apache.cassandra.io.util.File(diskFile));
+                }
+                catch(NoSuchFileException e)
+                {
+                    throw new RuntimeException(e);
+                }
+
+            default:
+                throw new IllegalStateException("Unknown backing media for ser.");
+        }
+    }
+
+    private static List<DeletionTime> generateDTs(int livePC)
```

Comment:
could we make it `livePercentage` and use it like percentage (i.e. `70` instead of `7` etc)

Reply:
Sure thing!

Reply:
Fixed in the latest commit already. Resolving but feel free to reopen if needed.